src/minimallist/generator.cljc

```diff
@@ -422,7 +422,8 @@
                                                  (gen/return (count required-entries)))
                                      selected-entries (gen/fmap (fn [shuffled-optional-entries]
                                                                   (->> (concat possible-entries shuffled-optional-entries)
-                                                                       (take coll-size)))
+                                                                       (take (-  coll-size (count required-entries)))
+                                                                       (concat required-entries)))
```

Comment:
I think that the real mistake is on the first line. It looks like it should be
`(concat required-entries shuffled-optional-entries)` instead of `(concat possible-entries shuffled-optional-entries)`.

PS: it's just a guess, I haven't tested it yet.

Reply:
**Update**: I did the test, and it solves the bug.